### ISSUE TITLE
Shared assets and classes between plugins

### ISSUE THREAD

**Issue by TheFOBas** [impresspages/ImpressPages#758]
I'm trying to create a couple of plugins that will share some assets and classes, but I wish to make them standalone. What is best way to make this happen? Where should I keep those classes and should I add them to different namespaces?

**Comment by maskas** [comment]
The namespace of each class must be the same as the folder that class appears in. Otherwise the class auto-loading won't work. You can always create a class from the other plugin using $x = new \Plugin\Xxx\Class(....);

you can also access assets of another plugin. Shouldn't be technically difficult except if you use it too much. That could be a sign of poor plugin decoupling.

**Comment by TheFOBas** [comment]
Yes I understand that but in my case there is no reason for user to have both of the plugins. For now the best idea that I have is to use third plugin as library.

**Comment by maskas** [comment]
Yes. That's an option. Or you can try to use git submodules too.






*Mangirdas Skripka*
mangirdas@impresspages.org
+37067128095

www.impresspages.org

Become our fan on Facebook <http://www.facebook.com/ImpressPages> and
follow us on Twitter <http://www.twitter.com/ImpressPages>
